### PR TITLE
Add spouse marriage fields to pensions

### PR DESCRIPTION
**Pull request by jbalboni**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3195

Adds spouse marriage pages, pretty similarly to the regular marriage pages.

I also updated the spouse financials pages to have a proper married check and added some missing tests from the regular marriage pages.

This form is huge.

screen shot 2017-06-13 at 4 27 53 pm
screen shot 2017-06-13 at 4 28 21 pm
screen shot 2017-06-13 at 4 28 30 pm
screen shot 2017-06-13 at 4 29 07 pm
screen shot 2017-06-13 at 4 28 51 pm
screen shot 2017-06-13 at 4 28 59 pm

### CODE REVIEW

File: src/js/common/schemaform/FieldTemplate.jsx [hunk 1]
```diff
@@ -55,7 +55,10 @@ export default function FieldTemplate(props) {
     'usa-input-error form-error-date': isDateField && hasErrors
   });
 
-  return (schema.type === 'object' || schema.type === 'array' || (schema.type === 'boolean' && !uiSchema['ui:widget'])) && !showFieldLabel
+  return (schema.type === 'object'
```
Comment (jbalboni) [inline src/js/common/schemaform/FieldTemplate.jsx]: This has been bugging me.

**Review by cvalarida — APPROVED** [review]

File: src/js/pensions/config/form.js [hunk 2]
```diff
@@ -493,6 +531,202 @@ const formConfig = {
...
+              'ui:title': 'What is the reason you do not live with your spouse?',
+              'ui:required': form => form.liveWithSpouse === false,
+              'ui:options': {
+                expandUnder: 'liveWithSpouse',
+                expandUnderCondition: false
+              }
+            },
+            monthlySpousePayment: {
+              // TODO: bold contribute monthly
+              'ui:title': 'How much do you contribute monthly to your spouse’s support?',
```
Comment (jbalboni) [inline src/js/pensions/config/form.js]: Yes, I just either needed to change the extension here or put it in the helper file, which I didn't do on the first pass. I'll update it.

File: test/pensions/config/marriageInfo.unit.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { findDOMNode } from 'react-dom';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import ReactTestUtils from 'react-dom/test-utils';
+
+import { DefinitionTester, submitForm } from '../../util/schemaform-utils.jsx';
+import formConfig from '../../../src/js/pensions/config/form';
+
+function fillData(formDOM, id, value) {
```
Comment (jbalboni) [inline test/pensions/config/marriageInfo.unit.spec.jsx]: Yeah, makes sense.

Comment (jbalboni) [inline test/pensions/config/marriageInfo.unit.spec.jsx]: I added a helper and updated the marriage unit tests to use it.

**Comment by J-Quag** [comment]
This is great. I have one question, is there a way to say "Current marriage" or have a way of specifying that "First" marriage does in fact mean your first marriage. Though that could just have been me assuming I was going to enter the Current marriage first.

**Comment by goldenmeanie** [comment]
I think it should be a screenbreak after the has spouse been married before question.

- Marriage history array:
  - First
  - Second
  - Current
--screen break--

**Comment by J-Quag** [comment]
The screen break is there, but when going into the spousal section, the title of the section still shows "First Marriage" as opposed to "Spouse's First Marriage" 
I'm not sure if we wanted to differentiate with a reminder that you are now on the Spouse's information.

**Pull request merged**